### PR TITLE
[image_picker] Add `getRecentMedia` function for fetching recent media items.

### PR DESCRIPTION
**Pull request by BeMacized**
This PR adds a `getRecentMedia` function to the platform interface, for fetching the last X most recent images or videos from the gallery, e.g. for allowing apps to pick recent items from within their own app rather than having to go to the separate media picking activity. 

Relevant PR:
 - flutter/flutter#19563

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/image_picker/image_picker_platform_interface/lib/src/method_channel/method_channel_image_picker.dart [hunk 1]
```diff
@@ -199,6 +199,47 @@ class MethodChannelImagePicker extends ImagePickerPlatform {
     return path != null ? XFile(path) : null;
   }
 
+  @override
+  Future<List<XFile>?> getRecentMedia({
+    required RetrieveType type,
```
Comment (stuartmorgan) [inline packages/image_picker/image_picker_platform_interface/lib/src/method_channel/method_channel_image_picker.dart:204]: This is for lost data; we should make a new enum.

File: packages/image_picker/image_picker_platform_interface/lib/src/method_channel/method_channel_image_picker.dart [hunk 2]
```diff
@@ -199,6 +199,47 @@ class MethodChannelImagePicker extends ImagePickerPlatform {
...
+    double? maxImageWidth,
+    double? maxImageHeight,
+    int? imageQuality,
```
Comment (stuartmorgan) [inline packages/image_picker/image_picker_platform_interface/lib/src/method_channel/method_channel_image_picker.dart:207]: These three parameters can use the new `ImageOptions`. And the overall method should use a parameter object for future-proofing.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Overall this looks reasonable (I skimmed the iOS and Android versions as well to see how they fit together), but please make a combined PR so that we can do a complete holistic review per the standard process: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changing-federated-plugins

**Comment by Hixie** [comment]
@BeMacized is this still on your radar?

**Comment by Hixie** [comment]
Closing due to lack of activity.

**Pull request closed without merging**